src/thermo/Phase.cpp

```diff
@@ -256,42 +256,122 @@ std::string Phase::speciesSPName(int k) const
     return m_name + ":" + speciesName(k);
 }
 
+vector<std::string> Phase::defaultState() const
+{
+    if (isPure()) {
+        if (isCompressible()) {
+            return {"T", "density"};
+        } else {
+            return {"T", "P"};
+        }
+    } else {
+        if (isCompressible()) {
+            return {"T", "density", "Y"};
+        } else {
+            return {"T", "P", "Y"};
+        }
+    }
+}
+
+vector<std::string> Phase::fullStates() const
+{
+    if (isPure()) {
+        if (isCompressible()) {
+            return {"TD", "TP", "UV", "DP", "HP", "SP", "SV"};
+        } else {
+            return {"TP", "HP", "SP"};
+        }
+    } else {
+        if (isCompressible()) {
+            return {"TDX", "TDY", "TPX", "TPY", "UVX", "UVY", "DPX", "DPY",
+                    "HPX", "HPY", "SPX", "SPY", "SVX", "SVY"};
+        } else {
+            return {"TPX", "TPY", "HPX", "HPY", "SPX", "SPY"};
+        }
+    }
+}
+
+vector<std::string> Phase::partialStates() const
+{
+    if (isPure()) {
+        return {};
+    } else {
+        if (!isCompressible()) {
+            return {"TP", "HP", "SP"};
+        } else {
+            return {"TD", "TP", "UV", "DP", "HP", "SP", "SV"};
+        }
+    }
+}
+
 void Phase::saveState(vector_fp& state) const
 {
-    state.resize(nSpecies() + 2);
+    if (isPure()) {
+        state.resize(2);
+    } else {
+        state.resize(nSpecies() +2);
+    }
```

Comment:
Likewise, I don't think that this should be necessary. I don't think that having one extra column in the output is a problem.

Reply:
Changed to `nStates`, per @speth's suggestion